github/Rakefile

```diff
@@ -0,0 +1,84 @@
+require "yaml"
+require "json"
+require "octokit"
+
+def repos
+  @repos ||= YAML.load_file("data/repos.yml")
+end
+
+def client
+  Octokit.auto_paginate = true
+  @client ||= Octokit::Client.new(access_token: ENV.fetch("GITHUB_TOKEN"))
+end
+
+desc "Fetch the names of all GOV.UK repos from alphagov"
+task :fetch_repos do
+  repos = client.search_repositories("topic:govuk user:alphagov", per_page: 100).items.map(&:full_name)
+  File.write("data/repos.yml", YAML.dump(repos.sort))
+end
+
+desc "Configure the repos with correct settings and branch protection"
+task :configure_repos do
+  repos.map do |repo|
+    client.edit_repository(
+      repo,
+      allow_merge_commit: true,
+      allow_squash_merge: false,
+      allow_rebase_merge: false,
+    )
+
+    client.protect_branch(
+      repo,
+      "master",
+      {
+        enforce_admins: true,
+        required_status_checks: {
+          strict: false, # "Require branches to be up to date before merging"
+          contexts: ["continuous-integration/jenkins/branch"]
+        },
+        required_pull_request_reviews: {
+          dismiss_stale_reviews: false,
+        }
+      }
+    )
+  end
+end
+
+desc "Create the correct webhooks for all repos"
+task :configure_webhooks do
```

Comment:
Reckon this should also add the Jenkins build hook too?

Reply:
Yes, I'd like to do that in a separate PR though (I haven't figured out how to install an "Integration" web hook and I want to make sure this works first).